Sources/TuistGenerator/Generator/DerivedFileGenerator.swift

```diff
@@ -15,12 +15,18 @@ protocol DerivedFileGenerating {
     /// - Returns: A project that might have got mutated after the generation of derived files, and a
     ///     function to be called after the project generation to delete the derived files that are not necessary anymore.
     func generate(graph: Graphing, project: Project, sourceRootPath: AbsolutePath) throws -> (Project, () throws -> Void)
+    func generateSideEffects(graph: Graphing, project: Project, sourceRootPath: AbsolutePath) throws -> (Project, [GeneratedSideEffect])
```

Comment:
What's the difference between `generate` and `generateSideEffects`, shouldn't both be together?

Reply:
Will follow up to removing the duplicate methods as par of this PR - they were left in initially to get the project compiling to get a working version to run against the acceptance tests before doing a sweep of consolidations / removals :)